src/data/data.cc

```diff
@@ -59,13 +84,39 @@ void MetaInfo::LoadBinary(dmlc::Stream *fi) {
                      << Version::String(version) << " is no longer supported. "
                      << "Please process and save your data in current version: "
                      << Version::String(Version::Self()) << " again.";
-  CHECK(fi->Read(&num_row_, sizeof(num_row_)) == sizeof(num_row_)) << "MetaInfo: invalid format";
-  CHECK(fi->Read(&num_col_, sizeof(num_col_)) == sizeof(num_col_)) << "MetaInfo: invalid format";
-  CHECK(fi->Read(&num_nonzero_, sizeof(num_nonzero_)) == sizeof(num_nonzero_))
-      << "MetaInfo: invalid format";
+
+  uint64_t num_field;
+  CHECK(fi->Read(&num_field)) << "MetaInfo: invalid format";
+  CHECK(num_field >= kNumField)
```

Comment:
This is a fatal error, next warning won't be effective.

Reply:
No, it's actually asserting that `num_field` is at least 7. The warning is shown if `num_field` is strictly greater than 7.